datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/utils/OwnerUtilsTest.java

```diff
@@ -32,38 +32,64 @@ public void testIsOwnerEqualUrnOnly() throws URISyntaxException {
   }
 
   @Test
-  public void testIsOwnerEqualWithLegacyTypeIsNotConsidered() throws URISyntaxException {
+  public void testIsOwnerEqualWithLegacyTypeOnly() throws URISyntaxException {
 
     Urn technicalOwnershipTypeUrn = new Urn(TECHNICAL_OWNER_OWNERSHIP_TYPE_URN);
     Urn ownerUrn1 = new Urn("urn:li:corpuser:foo");
-    Owner owner1 = new Owner();
-    owner1.setOwner(ownerUrn1);
-    owner1.setType(OwnershipType.TECHNICAL_OWNER);
+    Owner ownerWithTechnicalOwnership = new Owner();
+    ownerWithTechnicalOwnership.setOwner(ownerUrn1);
+    ownerWithTechnicalOwnership.setType(OwnershipType.TECHNICAL_OWNER);
 
-    assertTrue(OwnerUtils.isOwnerEqual(owner1, ownerUrn1, technicalOwnershipTypeUrn));
+    assertTrue(OwnerUtils.isOwnerEqual(ownerWithTechnicalOwnership, ownerUrn1, technicalOwnershipTypeUrn));
 
-    Owner owner2 = new Owner();
-    owner2.setOwner(ownerUrn1);
-    owner2.setType(OwnershipType.BUSINESS_OWNER);
-    assertTrue(
-        OwnerUtils.isOwnerEqual(owner1, ownerUrn1, new Urn(TECHNICAL_OWNER_OWNERSHIP_TYPE_URN)));
+    Owner ownerWithBusinessOwnership = new Owner();
+    ownerWithBusinessOwnership.setOwner(ownerUrn1);
+    ownerWithBusinessOwnership.setType(OwnershipType.BUSINESS_OWNER);
+    assertFalse(
+        OwnerUtils.isOwnerEqual(ownerWithBusinessOwnership, ownerUrn1, new Urn(TECHNICAL_OWNER_OWNERSHIP_TYPE_URN)));
   }
 
   @Test
-  public void testIsOwnerEqualOwnershipTypeUrnIsConsidered() throws URISyntaxException {
+  public void testIsOwnerEqualOnlyOwnershipTypeUrn() throws URISyntaxException {
 
     Urn technicalOwnershipTypeUrn = new Urn(TECHNICAL_OWNER_OWNERSHIP_TYPE_URN);
+    Urn businessOwnershipTypeUrn = new Urn(BUSINESS_OWNER_OWNERSHIP_TYPE_URN);
     Urn ownerUrn1 = new Urn("urn:li:corpuser:foo");
-    Owner owner1 = new Owner();
-    owner1.setOwner(ownerUrn1);
-    owner1.setTypeUrn(technicalOwnershipTypeUrn);
 
-    assertTrue(OwnerUtils.isOwnerEqual(owner1, ownerUrn1, technicalOwnershipTypeUrn));
+    Owner ownerWithTechnicalOwnership = new Owner();
+    ownerWithTechnicalOwnership.setOwner(ownerUrn1);
+    ownerWithTechnicalOwnership.setTypeUrn(technicalOwnershipTypeUrn);
+
+    Owner ownerWithBusinessOwnership = new Owner();
+    ownerWithBusinessOwnership.setOwner(ownerUrn1);
+    ownerWithBusinessOwnership.setTypeUrn(businessOwnershipTypeUrn);
+
+    Owner ownerWithoutOwnershipType = new Owner();
+    ownerWithoutOwnershipType.setOwner(ownerUrn1);
+
+    assertTrue(OwnerUtils.isOwnerEqual(ownerWithTechnicalOwnership, ownerUrn1, technicalOwnershipTypeUrn));
+    assertFalse(OwnerUtils.isOwnerEqual(ownerWithBusinessOwnership, ownerUrn1, technicalOwnershipTypeUrn));
+    assertFalse(OwnerUtils.isOwnerEqual(ownerWithTechnicalOwnership, ownerUrn1, null));
+    assertTrue(OwnerUtils.isOwnerEqual(ownerWithoutOwnershipType, ownerUrn1, null));
```

Comment:
We are missing 1 test here I think:

```java
assertTrue(OwnerUtils.isOwnerEqual(ownerWithoutOwnershipType, ownerUrn1, technicalOwnershipTypeUrn));
```

An owner without ownership type should be **equal** to the same owner being assigned a concrete ownership type so that all owners have a type of ownership assigned.

Reply:
Actually I don't think this case is possible since the deprecated `type` is mandatory in our pdl definition.

However we can have:
```java
Owner ownerWithoutOwnershipType = new Owner();
ownerWithoutOwnershipType.setOwner(ownerUrn1);
ownerWithoutOwnershipType.setType(NONE);

assertTrue(OwnerUtils.isOwnerEqual(ownerWithoutOwnershipType, ownerUrn1, technicalOwnershipTypeUrn));
```